purple-pill/js/controllers/mastersController.js

```diff
@@ -84,12 +85,53 @@ app.controller("mastersController", function($scope, sharedScope) {
         }
     };
 
-    $scope.importUfo = function() {
-        var message = "Want to load your own UFO?<br><br>Show us you want this by buying a T shirt:<br><ul><li><a title='Support the project and buy a T shirt (USA)' href='http://teespring.com/metapolator-beta-0-3-0' target='_blank' class='newtab'>USA</a></li><li><a title='Support the project and buy a T shirt (Worldwide)' href='http://metapolator.spreadshirt.com' target='_blank' class='newtab'>Worldwide</a></li>";
-        $scope.data.dialog(message, false, "close");
+    $scope.importUfo_dialog_open = function() {
+        $("#importufo_dialog").css("display", "block");
         $scope.data.localmenu.masters = false;
     };
 
+    $scope.handleUFOimportFiles = function(element) {
+        var ufozipfile = element.files[0]
+          , reader = new FileReader()
+          ;
+        $scope.data.dialog("Importing UFO ZIP...", true);
+        reader.onload = function(e) {
+            var importedMasters = $scope.data.stateful.project.importZippedUFOMasters(e.target.result)
+              , i, l
+              , referenceMaster = $scope.data.sequences[0].masters[0] //Should I really be doing it this way?
+              ;
+
+            for (i=0, l=importedMasters.length; i<l; i++){
+                var masterName = importedMasters[i]
+                  , model = $scope.data.stateful.project.open(masterName)
+                  , master = model.query('master#' + masterName)
+                  ;
+                $scope.data.sequences[0].masters.push({
+                    id : ++$scope.uniqueMasterId,
+                    name : masterName,
+                    displayName : "Master " + $scope.uniqueMasterId,
+//TODO:                    cpsFile : cpsFile,
+//TODO:                    ruleIndex : angular.copy(referenceMaster.ruleIndex),
```

Comment:
maybe we can solve this together with @jeroenbreen because I did not write this :-)
I guess:

`cpsFile`: the cps file that is used as the UI-Master "interface", i.e. a file like: https://github.com/metapolator/metapolator/blob/gh-pages/purple-pill/project/data/com.metapolator/cps/lib/parameters.cps but with a different definition for ```* {
    baseMaster: S"master#base";
}``` (what is done at the moment in https://github.com/metapolator/metapolator/blob/gh-pages/purple-pill/project/data/com.metapolator/cps/lib/flexmaster.cps)


`ruleIndex` you need to know where in the paramterCollection (that is created by parsing a CPS-file) the rule you wan't to change is located. I don't know which rules' index is meant here exactly.